src/libsyntax_ext/deriving/custom.rs

```diff
@@ -54,7 +54,7 @@ impl MultiItemModifier for CustomDerive {
             Annotatable::Item(item) => item,
             Annotatable::ImplItem(_) |
             Annotatable::TraitItem(_) => {
-                ecx.span_err(span, "custom derive attributes may only be \
+                ecx.span_err(span, "proc_macro_derive attributes may only be \
```

Comment:
"proc_macro_derive attribute" makes me think of `#[proc_macro_derive]` itself rather than `#[derive(Trait)]`.
Maybe "proc-macro derives may only be ..."?